src/PerfView/Dialogs/RunCommandDialog.xaml.cs

```diff
@@ -295,7 +302,25 @@ private void OKButtonClick(object sender, RoutedEventArgs e)
                 }
 
                 if (!m_isCollect)
+                {
                     m_args.CommandLine = CommandToRunTextBox.Text;
+
+                    if (CommandToRunTextBox.AddToHistory(m_args.CommandLine))
+                    {
+                        StringBuilder sb = new StringBuilder();
+                        foreach (string item in CommandToRunTextBox.Items)
+                        {
+                            if (item != "")
+                            {
```

Comment:
I worry if the command had a ; in it (although that is weird).  rather than implementing an escape mechanism, I would recommend simply putting !item.Contains(';') (thus we don't store in history things with ';').